src/esp/assets/ResourceManager.cpp

```diff
@@ -1108,6 +1109,263 @@ bool ResourceManager::loadGeneralMeshData(
   return true;
 }  // loadGeneralMeshData
 
+std::vector<Mn::Vector3> ResourceManager::buildSmoothTrajOfPoints(
+    const std::vector<Mn::Vector3>& pts,
+    int numInterp) {
+  std::vector<Mn::Math::CubicHermite<Mn::Vector3>> splinePath;
+  std::vector<Mn::Vector3> trajectory;
+  Mn::Vector3 a, b;
+  double tStep = 1.0 / numInterp;
+  // pts.size() must be > 1
+  // beginning point - tangents are opposites
+  splinePath.emplace_back(buildSpline({(pts[0] - pts[1]).normalized()},
+                                      {(pts[1] - pts[0]).normalized()},
+                                      pts[0]));
+  int numPtsM1 = pts.size() - 1;
+  for (int i = 1; i < numPtsM1; ++i) {
+    splinePath.emplace_back(buildSpline({(pts[i] - pts[i - 1]).normalized()},
+                                        {(pts[i + 1] - pts[i]).normalized()},
+                                        pts[i]));
+
+    for (double t = 0.0; t < 1.0; t += tStep) {
+      trajectory.emplace_back(
+          Mn::Math::splerp(splinePath[i - 1], splinePath[i], t));
+    }
+  }
+
+  // end point - tangents are opposites
+  splinePath.emplace_back(buildSpline(
+      {(pts[numPtsM1] - pts[numPtsM1 - 1]).normalized()},
+      {(pts[numPtsM1 - 1] - pts[numPtsM1]).normalized()}, pts[numPtsM1]));
+  for (double t = 0.0; t < 1.0; t += 0.025) {
+    trajectory.emplace_back(
+        Mn::Math::splerp(splinePath[numPtsM1 - 1], splinePath[numPtsM1], t));
+  }
+  return trajectory;
+}  // ResourceManager::buildSmoothTrajOfPoints
+
+Mn::Trade::MeshData ResourceManager::trajectoryTubeSolid(
+    const std::vector<Mn::Vector3>& pts,
+    int numSegments,
+    int numInterp,
+    float radius) {
+  // 1. Build smoothed trajectory through passed points
+  std::vector<Mn::Vector3> trajectory =
+      pts;  // buildSmoothTrajOfPoints(pts, numInterp);
```

Comment:
This comment is confusing.

Reply:
I think this is just an optional step to smooth the provided trajectory with spline interpolation and is commented out here.

Reply:
Thank you for checking things out.  This is a very early WIP, here so that mosra can help me debug the weird striding issue with the vert interleaving.  Please keep that in mind as you look over things.